dist_tables/querying.rst

```diff
@@ -76,16 +76,48 @@ You can then call sum(hll_column) to roll up those columns within the database.
+  EXPLAIN
+  SELECT location_id, count(driver_id)
+  FROM deliveries
+  GROUP BY location_id
+  LIMIT 5;
+
+::
+
+  ┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+  │                                                           QUERY PLAN                                                             │
+  ├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
+  │ Limit  (cost=0.00..0.00 rows=0 width=0)                                                                                          │
+  │   ->  HashAggregate  (cost=0.00..0.00 rows=0 width=0)                                                                            │
+  │     Group Key: remote_scan.worker_column_2                                                                                       │
+  │     ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)                                                          │
+  │       Task Count: 32                                                                                                             │
+  │       Tasks Shown: One of 32                                                                                                     │
+  │       ->  Task                                                                                                                   │
+  │         Node: host=localhost port=5433 dbname=postgres                                                                           │
+  │         ->  Limit  (cost=0.15..0.64 rows=5 width=16)                                                                             │
+  │           ->  GroupAggregate  (cost=0.15..69.15 rows=700 width=16)                                                               │
+  │             Group Key: location_id                                                                                               │
+  │             ->  Index Only Scan using deliveries_pkey_102360 on deliveries_102360 companies  (cost=0.15..58.65 rows=700 width=8) │
+  └──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+  (12 rows)
+
+Notice how the limit occurs twice in the plan: once at the final step in the coordinator, and also down in the query run on the workers. This limits how much data each worker returns to the coordinator.
+
+However, in some cases SELECT queries with LIMIT clauses may need to fetch all rows from each shard to generate exact results. For example, if we had added an ORDER BY the driver count then the coordinator would need results from all shards to determine the final aggregate value. No individual worker knows which rows are safe to omit, and would send back all rows. This reduces performance of the LIMIT clause due to high volume of network data transfer.
```

Comment:
Actually I didn't quite follow the logic of why limits cannot be pushed down when ordering by the aggregate. For example wouldn't the five  locations with most drivers be chosen from among the best five from each shard? A sixth place location on shard 23 has no chance in the global competition, right?

I must have not parsed the claim properly when rewriting the paragraph.

Reply:
@begriffs Let's assume the top-5 in each shard are different. However, the 6th place might be the same across all shards (i.e., just miss out on making top-5). However, in aggregate across all shards, it might come out on top. 

Is there anything remaining on this PR btw? I had approved this one.